pkg/hooks/manager.go

```diff
@@ -161,15 +167,11 @@ func processSideCarSocket(socketPath string) (*callBackClient, bool, error) {
 		versionsSet[version] = true
 	}
 
-	// The order matters. We should match newer versions first.
-	supportedVersions := []string{
-		hooksV1alpha3.Version,
-		hooksV1alpha2.Version,
-		hooksV1alpha1.Version,
-	}
-
-	for _, version := range supportedVersions {
+	for version, isDeprecated := range SupportedVersions {
```

Comment:
Just to clarify: previously, it was `The order matters. We should match newer versions first.`. Now we iterate over a map here and hence the order of elements is randomized. Wouldn't it cause issues?

Reply:
Yes, you are right. InfoResult reports an array of [versions](https://github.com/kubevirt/kubevirt/blob/main/pkg/hooks/info/api_info.pb.go#L56). On manager.go, we should prefer the latest version.

Many thanks for catching this.